[T282860.2] Add api-testing

toban opened the pull request:
Adds api-testing and two simple tests describing this scenario https://phabricator.wikimedia.org/T282858#7095398

https://phabricator.wikimedia.org/T282860

toban committed:
Add mediawiki api-testing

This adds the mediawiki api-testing framework and a simple test.

Bug: T282860

lucaswerkmeister commented on tests/api-testing/REST/EditEndpoint.js:
```diff
@@ -0,0 +1,106 @@
+		// could use PropertyLabelResolver to go around knowing the P numbers
+		reconciliationPropertyId = await requestHelper.getOrCreateProperty( 'url', 'identifier' );
+		namePropertyId = await requestHelper.getOrCreateProperty( 'string', 'name' );
+		billOfMaterialPropertyId = await requestHelper.getOrCreateProperty( 'wikibase-item', 'Bill of material' );
```
As far as I can tell the common spelling is “bill of material**s**”, not “material”. (Also, I see no reason to write “bill” in uppercase.)

toban replied:
done!

lucaswerkmeister reviewed (commented):
Looks pretty good, only a minor nitpick so far. (Haven’t tried it out yet. Also, this could probably be rebased on `main` now that #29 is merged.)

toban committed:
Add linting job and run it

toban commented:
> Looks pretty good, only a minor nitpick so far. (Haven’t tried it out yet. Also, this could probably be rebased on `main` now that #29 is merged.)

Fixed the copy things but also realized the linter wasn't running so added a commit for that.

lucaswerkmeister commented on .github/workflows/ci.yml:
```diff
@@ -15,6 +15,15 @@ jobs:
+            - uses: actions/checkout@v2
+            - name: lint
+              run: |
+                  touch .api-testing.config.json
```
Hm, I feel like it would make more sense to resolve the `node/no-missing-require` error with an `eslint-disable-line` directive, so that running the linter also works locally. without creating that config file.

toban replied:
yup sounds good!

lucaswerkmeister commented on composer.json:
```diff
@@ -13,7 +13,7 @@
 		"test": [
 			"parallel-lint . --exclude vendor --exclude node_modules",
-			"phpcs -p -s",
+			"phpcs -p -s  --ignore='tests/api-testing'",
```
Hm, is this necessary? I assume we shouldn’t have any PHP files in that directory anyways.

toban replied:
If you look [lint run from previous commit](https://github.com/wmde/WikibaseReconcileEdit/runs/2622625960), It's annoyingly picking up the requestHelper.js

lucaswerkmeister replied:
WTF. ok

toban committed:
[T282860.3] Run api tests on ci (#28)

This commit adds another job for running the api-testing specs against a
Wikibase instance with WikibaseReconcileEdit enabled.

This currently only runs against 1.35.2-wmde.1 but could with some
modifications run from a matrix of releases.
